### PR TITLE
Cleanup pascal code

### PR DESCRIPTION
**Pull request by turric4n**

### CODE REVIEW

**Comment by turric4n** [comment]
Improve Pascal performance.

File: fibonacci/fpc/code.pas [hunk 1]
```diff
@@ -3,26 +3,15 @@
 uses
   SysUtils;
 
-function Fibonacci(n: Int32): Int32;
+function Fibonacci(n: UInt32): UInt32; inline;
```
Comment (bddicken) [inline fibonacci/fpc/code.pas:6]: What does `inline` do here?

File: fibonacci/fpc/code.pas [hunk 2]
```diff
@@ -3,26 +3,15 @@
...
 begin
-  if n = 0 then
-    Fibonacci := 0
-  else if n = 1 then
-    Fibonacci := 1
-  else
-    Fibonacci := Fibonacci(n - 1) + Fibonacci(n - 2);
+  if n <= 1 then exit(n);
```
Comment (bddicken) [inline fibonacci/fpc/code.pas:8]: Let's revert this back to the if/else.

File: loops/fpc/code.pas [hunk 3]
```diff
@@ -4,28 +4,20 @@
   SysUtils;
 
 var
-  u, r, i, j: Int32;
-  a: array[0..9999] of Int32;
+  u: UInt32;
+  a: array[0..9999] of UInt32;
```
Comment (bddicken) [inline loops/fpc/code.pas:8]: Interesting! Does using a unit give better performance than a signed iint?

Comment (turric4n) [reply loops/fpc/code.pas:8]: Surprisingly, yes, it's 40% more efficient.

File: fibonacci/fpc/code.pas (hunk 1 above)
Comment (turric4n) [reply fibonacci/fpc/code.pas:6]: Procedures that are declared inline are copied to the places where they are called. This has the effect that there is no actual procedure call, the code of the procedure is just copied to where the procedure is needed, this results in faster execution speed if the function or procedure is used a lot. It is obvious that inlining large functions does not make sense.

File: fibonacci/fpc/code.pas (hunk 2 above)
Comment (turric4n) [reply fibonacci/fpc/code.pas:8]: ok,  no problem

**Comment by bddicken** [comment]
Nice, thanks!

**Pull request merged**